web/src/components/User.js

```diff
@@ -1,130 +1,263 @@
+    transform: rotate(45deg) translate(0px, -12px);
+  }
 `;
 
 const Actions = styled.div`
-  > button {
-    margin-bottom: 27px;
-  }
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 20px;
 `;
 
-const ActionButton = styled(Button)`
-  font-size: 16px;
-  background: #F0F0F0 0% 0% no-repeat padding-box;
-  width: 100%;
-  color: #696969;
-  border-width: 0px;
+const ActionButton = styled.div`
+  width: 146px;
+  height: 48px;
+  box-sizing: border-box;
+  background: #FFFFFF;
+  border-radius: 24px;
+  box-shadow: 0px 3px 6px #00000080;
+  text-align: center;
+  font-size: 20px;
+  font-weight: bolder;
+  color: #000000;
+  padding: 10px 17px;
+  cursor: pointer;
+`;
+
+const UserWide = styled.div`
+  display: none;
 
-  a:visited {
-    color: #696969;
+  @media(min-width: 1920px) {
+    position: fixed;
+    left: 65%;
+    display: block;
+    min-width: 380px;
+    padding: 28px 32px 0 32px;
+    background: #0F2250 0% 0% no-repeat padding-box;
+    box-shadow: 0px 3px 6px #1B3A76B5;
+    border-radius: 39px;
+  }
+
+  ${Username} {
+    margin-bottom: 10px;
+  }
+
+  ${UserSignout} {
+    margin: 0;
+  }
+
+  ${Points} {
+    margin-top: 30px;
+    margin-bottom: 20px;
   }
+
+  ${ActionButton} {
+    width: 48%;
+  }
+
 `;
 
+const User = ({nickname, points }) => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
 
-const User = ({nickname, points, imageSrc}) => {
+  const openMenu = () => {
+    setIsMenuOpen(!isMenuOpen);
+  }
 
-  const handleTrading = () => {
-    console.log('trading click');
+  const handleMask = () => {
+    document.getElementById("menuCheck").checked = false;
+    setIsMenuOpen(!isMenuOpen);
   }
 
   return (
     <>
+    <PositionFixed>
     <Container>
+      {
+        isMenuOpen ?
+        <Menu>
+          <Mask onClick={handleMask} />
+          <Points>
+             現有點數：{points} P
+          </Points>
+          <Actions>
+            <ActionButton>交易點數</ActionButton>
+            <ActionButton>兌換點數</ActionButton>
+          </Actions>
+        </Menu> :
+        null
+      }
       <UserInfo>
-        <div className="flex-column">
-          <Username>{nickname}</Username>
-          <UserSignout href="">sign out</UserSignout>
-        </div>
-        <div>
-          <UserImage src={imageSrc}  onClick="codePopUp()" />
-        </div>
+        <Username>
+          {nickname}
+          {
+            isMenuOpen ?
+            <UserSignout href="">sign out</UserSignout> :
+            null
+          }
+        </Username>
+        <ActionBar onClick={openMenu}>
+          <input type="checkbox" id="menuCheck"/>
+          <span />
+          <span />
+          <span />
+        </ActionBar>
       </UserInfo>
-
-      <PointWrapper>
-        <Quantity>{points}</Quantity>
-        <Unit>P</Unit>
-      </PointWrapper>
-
+    </Container>
+    </PositionFixed>
+    <UserWide>
+      <Username>{nickname}</Username>
+      <UserSignout href="">sign out</UserSignout>
```

Comment:
Remove sign out. We use token-based and it doesn't like username/password have the sign-out function.

Reply:
got It